packages/vc-verification/src/role-definition-resolver.ts

```diff
@@ -0,0 +1,26 @@
+import { CachedRoleDefinitionResolver, ENSRoleDefinitionResolver } from '.';
+export class RoleDefinitionResolver {
+  private _roleDefResolver:
+    | CachedRoleDefinitionResolver
+    | ENSRoleDefinitionResolver;
+
+  /**
+   * The RoleDefinitionResolver class could be initialised either with
+   * CachedRoleDefinitionResolver or ENSRoleDefinitionResolver
+   * @param rolDefResolver
+   */
+  constructor(
+    rolDefResolver: CachedRoleDefinitionResolver | ENSRoleDefinitionResolver
+  ) {
+    this._roleDefResolver = rolDefResolver;
+  }
```

Comment:
Would clients need to modify this code in order to add a new implementation?
I'm wondering if maybe it would be better if `RoleDefinitionResolver` were an interface so that a client (e.g. ssi-hub`) can add their own implementation without needing to modify this code.